SwiftCompilerSources/Sources/Optimizer/FunctionPasses/ClosureSpecialization.swift

```diff
@@ -166,7 +166,16 @@ let autodiffClosureSpecialization = FunctionPass(name: "autodiff-closure-special
     var changed = false
     for inst in function.instructions {
       if let partialApply = inst as? PartialApplyInst,
-         partialApply.isPullbackInResultOfAutodiffVJP
+         partialApply.isPullbackInResultOfAutodiffVJP,
+         // Calling `cloneRecursively` from `SpecializationInfo.cloneClosures`
+         // requires the pullback having ownership info. Otherwise, the cloner
+         // uses `recordFoldedValue` instead of `recordClonedInstruction`, and
+         // `postProcess` hook is not called, which leads to an assertion
+         // failure in `BridgedClonerImpl::cloneInst`.
+         // TODO: investigate why the assertion fails only on Linux. See
+         // * https://github.com/swiftlang/swift/pull/84800#issuecomment-3398200989
+         // * https://github.com/swiftlang/swift/issues/84955
+         partialApply.referencedFunction?.hasOwnership == true
```

Comment:
Can you put this check into `isCalleeSpecializable`, because we need to check it also in the regular closure specialization

Reply:
> Can you put this check into `isCalleeSpecializable`, because we need to check it also in the regular closure specialization

Applied the suggestion in 93169c183e0f4dfca8b6f8df365087198028bcc1, thanks.